src/main/java/uk/gov/companieshouse/insolvency/data/serialization/LocalDateDeSerializer.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.companieshouse.insolvency.data.serialization;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+
+public class LocalDateDeSerializer extends JsonDeserializer<LocalDate> {
+
+    @Override
+    public LocalDate deserialize(JsonParser jsonParser, DeserializationContext
+            deserializationContext) throws IOException, JacksonException {
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter
+                .ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'");
+        JsonNode jsonNode = jsonParser.readValueAsTree();
+        try {
+            return LocalDate.parse(jsonNode.get("$date").textValue(), dateTimeFormatter);
+        } catch (Exception ex) {
+            //TODO: handle exception
```

Comment:
Please log and re-throw the exception to handle in retry/error stories

Reply:
Code change done.